test/module/irohad/model/command_validate_execute_test.cpp

```diff
@@ -212,6 +213,145 @@ TEST_F(AddAssetQuantityTest, InvalidWhenNoAsset) {
   ASSERT_FALSE(validateAndExecute());
 }
 
+class SubtractAssetQuantityTest : public CommandValidateExecuteTest {
+public:
+    void SetUp() override {
+      CommandValidateExecuteTest::SetUp();
+
+      asset = Asset();
+      asset.asset_id = asset_id;
+      asset.domain_id = domain_id;
+      asset.precision = 2;
+
+      wallet = AccountAsset();
+      wallet.asset_id = asset_id;
+      wallet.account_id = account_id;
+      wallet.balance = balance;
+
+      subtract_asset_quantity = std::make_shared<SubtractAssetQuantity>();
+      subtract_asset_quantity->account_id = creator.account_id;
+      Amount amount(100, 2);
+      subtract_asset_quantity->amount = amount;
+      subtract_asset_quantity->asset_id = asset_id;
+
+      command = subtract_asset_quantity;
+      role_permissions = {can_subtract_asset_qty};
+    }
+
+    decltype(AccountAsset().balance) balance = Amount(150ul, 2);
+    Asset asset;
+    AccountAsset wallet;
+
+    std::shared_ptr<SubtractAssetQuantity> subtract_asset_quantity;
+};
+
+TEST_F(SubtractAssetQuantityTest, InvalidWhenNoWallet) {
```

Comment:
We started using [gwt](https://en.wikipedia.org/wiki/Given-When-Then) way to write test docs, look at this for example:
https://github.com/hyperledger/iroha/blob/19651930fa6851586bfb9f09c3f6f2f3d3a51eaf/test/module/irohad/ametsuchi/block_query_test.cpp#L84-L90